src/tasks/tasks.py

```diff
@@ -426,6 +435,12 @@ def get_metrics(self, reset=False):
         return {'mcc': self.scorer1.get_metric(reset),
                 'accuracy': self.scorer2.get_metric(reset)}
 
+    def update_metrics(self, logits, labels, tagmask=None):
+        logits, labels = logits.detach(), labels.detach()
```

Comment:
Why the detach? (Don't think it's wrong, just not sure.)

Reply:
Pytorch sometimes reports errors when converting a gpu tensor to cpu, if it is needed to compute some other tensor that requires grad. Here, logits and labels are used to compute loss, while they will be moved to cpu within the scorer updating function.